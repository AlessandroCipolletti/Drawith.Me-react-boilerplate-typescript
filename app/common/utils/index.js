export function parseColorRgbStringToRgba (color) {
  color = color.substring(4).replace(")", "").replace(" ", "").split(",")
  return {
    r: parseInt(color[0]),
    g: parseInt(color[1]),
    b: parseInt(color[2]),
    a: color[3] ? parseInt(color[3]) : 1,
  }
}

export function parseColorHexToRgb (hex) {
  return {
    r: parseInt(hex.substring(0, 2), 16),
    g: parseInt(hex.substring(2, 4), 16),
    b: parseInt(hex.substring(4, 6), 16),
  }
}

export function parseColorRgbToHex (r, g, b) {
  const hex = ((1 << 24) + (r << 16) + (g << 8) + b).toString(16).slice(1)
  return `#${hex}`
}

export function parseColorHexToRgbaString (colorHex, alpha = 1) {
  const r = parseInt(colorHex.substr(1, 2), 16)
  const g = parseInt(colorHex.substr(3, 2), 16)
  const b = parseInt(colorHex.substr(5, 2), 16)
  return `rgba(${r}, ${g}, ${b}, ${alpha})`
}
