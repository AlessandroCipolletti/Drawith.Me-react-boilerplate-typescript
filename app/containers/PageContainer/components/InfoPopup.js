import React from 'react'
import PropTypes from 'prop-types'

import Popup from 'common/components/Popup'

const InfoPopup = ({
  callback,
}) => (
  <Popup callback={callback}>
    ciao ciao
  </Popup>
)

InfoPopup.propTypes = {
  callback: PropTypes.func.isRequired,
}

export default InfoPopup
