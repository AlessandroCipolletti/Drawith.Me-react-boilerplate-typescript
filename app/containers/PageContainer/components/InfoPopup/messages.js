import { defineMessages } from 'react-intl'

export const scope = 'app.components.InfoPopup'

export default defineMessages({
  title: {
    id: `${scope}.title`,
    defaultMessage: 'Share your creativity',
  },

  descriptionLine1: {
    id: `${scope}.description_line_1`,
    defaultMessage: 'Created by',
  },
  descriptionLine2: {
    id: `${scope}.description_line_2`,
    defaultMessage: 'Alessandro Cipolletti',
  },
  descriptionLine3: {
    id: `${scope}.description_line_3`,
    defaultMessage: 'cipolletti.alessandro@gmail.com',
  },
})
