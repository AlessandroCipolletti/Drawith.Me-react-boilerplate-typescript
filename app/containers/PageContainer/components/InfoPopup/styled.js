import styled from 'styled-components'

import headerBG from 'images/UI/PageContainer/headerBG.png'
import logo from 'images/UI/PageContainer/logo.png'

export const Wrapper = styled.div`
  position: relative;
  width: ${props => props.theme.spacing.size15};
  height: ${props => props.theme.spacing.size15};
`

export const Header = styled.div`
  position: absolute;
  top: 0px;
  left: 0px;
  width: 100%;
  height: ${props => props.theme.spacing.size9};
  border-bottom: solid 1px ${props => props.theme.palette.gray[6]};
  background-image: url(${logo}), url(${headerBG});
  background-size: auto 90%, contain;
  background-repeat: no-repeat, repeat-x;
  background-position: center, 0% 0%;
`

export const Content = styled.div`
  position: relative;
  width: 100%;
  height: 100%;
  box-sizing: border-box;
  padding: ${props => props.theme.spacing.size4};
  padding-top: ${props => `${parseInt(props.theme.spacing.size9) + parseInt(props.theme.spacing.size4)}px`};
  text-align: center;
`
