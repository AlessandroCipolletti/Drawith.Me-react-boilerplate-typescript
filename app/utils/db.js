import Dexie from 'dexie'
// /* eslint-disable no-unused-vars */
// import setGlobalVars from 'indexeddbshim'
// setGlobalVars()

const db = new Dexie('DrawithMe.drawings')
db.version(1).stores({
  user: 'id, publicName, firstName, lastName',
  drawings: '++id, userId, folderId, state, createTime, updateTime, previewPath, imagePath, minX, minY, maxX, maxY, width, height',
  folders: '++id, name, order',
})

export default db
