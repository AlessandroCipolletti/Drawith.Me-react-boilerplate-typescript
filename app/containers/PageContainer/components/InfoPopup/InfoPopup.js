import React from 'react'
import PropTypes from 'prop-types'
import { compose } from 'redux'
import { injectIntl, intlShape } from 'react-intl'

import Popup from 'common/components/Popup'
import { Wrapper, Header, Content, Title, Image, Description } from './styled'

import messages from './messages'

const InfoPopup = ({
  callback, intl,
}) => (
  <Popup callback={callback}>
    <Wrapper>
      <Header />
      <Content>
        <Title>{intl.formatMessage(messages.title)}</Title>
        <Image />
        <Description>
          <br /><br />
          {intl.formatMessage(messages.descriptionLine1)}
          <br />
          <big>{intl.formatMessage(messages.descriptionLine2)}</big>
          <br /><br />
          {intl.formatMessage(messages.descriptionLine3)}
        </Description>
      </Content>
    </Wrapper>
  </Popup>
)

InfoPopup.propTypes = {
  callback: PropTypes.func.isRequired,
  intl: intlShape.isRequired,
}

export default compose(
  injectIntl,
)(InfoPopup)
