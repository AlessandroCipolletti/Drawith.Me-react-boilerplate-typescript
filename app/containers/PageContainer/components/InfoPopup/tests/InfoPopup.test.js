import React from 'react'
import { render } from 'react-testing-library'

// Import de Intl Provider
import { IntlProvider, injectIntl } from 'react-intl'
import { compose } from 'redux'

// Import de Theme Provider
import { ThemeProvider } from 'styled-components'
import Theme from 'common/Theme'
import { DEFAULT_LOCALE } from 'utils/i18n'

import InfoPopup from '../InfoPopup'

describe('<InfoPopup />', () => {

  const ComponentWithIntl = compose(injectIntl)(InfoPopup)
  const renderComponent = (props = {}) => (
    render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <ThemeProvider theme={Theme}>
          <ComponentWithIntl {...props} />
        </ThemeProvider>
      </IntlProvider>,
    )
  )

  afterEach(() => {
    jest.clearAllMocks()
  })

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error')
    const callback = () => {}

    renderComponent({ callback })

    expect(spy).not.toHaveBeenCalled()
  })

  it('Expect log an error in console if called without callback prop', () => {
    const spy = jest.spyOn(global.console, 'error')

    renderComponent()

    expect(spy).toHaveBeenCalledTimes(2)
  })

  it('Should render and match the snapshot with default props', () => {
    const callback = () => {}

    const {
      container: { firstChild },
    } = renderComponent({ callback })

    expect(firstChild).toMatchSnapshot()
  })
})
