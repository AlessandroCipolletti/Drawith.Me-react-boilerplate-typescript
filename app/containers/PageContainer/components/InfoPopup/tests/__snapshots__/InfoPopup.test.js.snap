// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfoPopup /> Should render and match the snapshot with default props 1`] = `null`;
